TM1py/Services/ElementService.py

```diff
@@ -142,7 +141,27 @@ def get_elements_by_level(self, dimension_name: str, hierarchy_name: str, level:
             "/api/v1/Dimensions('{}')/Hierarchies('{}')/Elements?$select=Name&$filter=Level eq {}",
             dimension_name,
             hierarchy_name,
-            level)
+            str(level))
+        response = self._rest.GET(url, **kwargs)
+        return [e["Name"] for e in response.json()['value']]
+
+    def get_elements_by_wildcard(self, dimension_name: str, hierarchy_name: str,
+                                 wildcard: str, level: int = None, **kwargs) -> List[str]:
+        """ Get all element names filtered by wildcard (CaseInsensitive) and level in a hierarchy
+
+        :param dimension_name: Name of the dimension
+        :param hierarchy_name: Name of the hierarchy
+        :param wildcard: wildcard to filter
+        :param level: Level to filter
+        :return: List of element names
+        """
+        filter_elements = format_url("contains(tolower(Name),tolower('{}'))", wildcard)
```

Comment:
2 things about this filter:

- we have to be careful, not to apply `format_url` on a string twice. I think if you are using it below, you can skip it here.
- If it's not too expensive, I think it would be nice to cater to space insensitivity as well: `tolower(replace(Name, ' ', ''))`

Reply:
Performs _CaseAndSpaceInsensitive_ filter now.